neqo-transport/src/fc.rs

```diff
@@ -318,6 +355,34 @@ impl ReceiverFlowControl<StreamId> {
             }
         }
     }
+
+    pub fn add_retired(&mut self, count: u64) {
+        debug_assert!(self.retired + count <= self.consumed);
+        self.retired += count;
+        if self.retired + self.max_active / 2 > self.max_allowed {
+            self.frame_pending = true;
+        }
+    }
+
+    pub fn set_consumed(&mut self, consumed: u64) -> Res<u64> {
+        if consumed > self.consumed {
+            if consumed > self.max_allowed {
+                qtrace!("Stream RX window exceeded: {}", consumed);
+                return Err(Error::FlowControlError);
+            }
+            let new_consumed = consumed - self.consumed;
+            self.consumed = consumed;
+            Ok(new_consumed)
+        } else {
+            Ok(0)
+        }
```

Comment:
maybe you can just do: 
```rust
    let x = consumed.saturating_sub(self.consumed); 
    self.consume(x)?; 
    Ok(x)
```

Reply:
his does not work because consume is not implemented for ReceiverFlowControl<StreamId> because it is not needed. ReceiverFlowControl<StreamId> only uses set_consumed.

Reply:
Flip it the other way around then.